include/dd/Package.hpp

```diff
@@ -1672,7 +1794,8 @@ namespace dd {
         /// Noise Operations
         ///
     public:
-        NoiseOperationTable<mEdge> noiseOperationTable{nqubits};
+        StochasticNoiseOperationTable<mEdge>                  stochasticNoiseOperationCache{nqubits};
```

Comment:
Is there any chance we could avoid allocating this table if it is not needed?
I would assume it takes up quite some space for it to not be used besides stochastic noise simulation.

Reply:
`StochasticNoiseOperationTable<mEdge>` now takes an additional template parameter `size`, which allows setting the size when initializing the ddPackage. This also eliminates the magic number.